src/ucs/datastruct/static_bitmap.h

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2023. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCS_STATIC_BITMAP_H_
+#define UCS_STATIC_BITMAP_H_
+
+#include "bitmap.h"
+
+#include <ucs/debug/assert.h>
+#include <ucs/sys/math.h>
+#include <ucs/sys/preprocessor.h>
+
+BEGIN_C_DECLS
+
+#define ucs_static_bitmap_s(_num_bits) \
+    struct { \
+        ucs_bitmap_word_t \
+                bits[ucs_div_round_up(_num_bits, UCS_BITMAP_BITS_IN_WORD)]; \
+    }
+
+
+#define UCS_STATIC_BITMAP_ZERO_INITIALIZER \
+    { \
+        .bits = { 0 } \
+    }
+
+
+/**
+ * Get the number of words in a given bitmap
+ *
+ * @param _bitmap Bitmap variable to get words number
+ *
+ * @return Number of words
+ */
+#define UCS_STATIC_BITMAP_NUM_WORDS(_bitmap) \
+    ucs_static_array_size((_bitmap).bits)
+
+
+#define UCS_STATIC_BITMAP_BITS_ARGS(_bitmap_ptr) \
+    ((ucs_bitmap_word_t*)(_bitmap_ptr)->bits), \
+            UCS_STATIC_BITMAP_NUM_WORDS(*(_bitmap_ptr))
+
+
+#define UCS_STATIC_BITMAP_BITS_CARGS(_bitmap_ptr) \
+    ((const ucs_bitmap_word_t*)(_bitmap_ptr)->bits), \
+            UCS_STATIC_BITMAP_NUM_WORDS(*(_bitmap_ptr))
+
+
+#define _UCS_STATIC_BITMAP_FUNC_CALL(_uid, _bitmap, _func, ...) \
```

Comment:
`uid` like "user id"? Why?

Reply:
like unique id, to create a temp variable name that will be different from others